cmd/system-probe/subcommands/ebpf/command_linux.go

```diff
@@ -189,7 +193,112 @@ func isPerCPUMap(mapType ebpf.MapType) bool {
 		mapType == ebpf.LRUCPUHash
 }
 
+// bpfMapInfo mirrors the kernel's bpf_map_info structure fields we need
+type bpfMapInfo struct {
+	mapType                uint32
+	id                     uint32
+	keySize                uint32
+	valueSize              uint32
+	maxEntries             uint32
+	mapFlags               uint32
+	name                   [16]byte
+	ifindex                uint32
+	btfVmlinuxValueTypeID  uint32
+	netnsDev               uint64
+	netnsIno               uint64
+	btfID                  uint32
+	btfKeyTypeID           uint32
+	btfValueTypeID         uint32
+	btfVmlinuxIDUnused     uint32
+	mapExtra               uint64
+}
+
+// getBTFTypeIDsFromSyscall directly calls BPF_OBJ_GET_INFO_BY_FD to get BTF type IDs
+func getBTFTypeIDsFromSyscall(m *ebpf.Map) (btfID uint32, keyTypeID btf.TypeID, valueTypeID btf.TypeID, err error) {
```

Comment:
Why are you making raw syscalls, and not using the cilium/ebpf API here as well?